openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -726,14 +730,31 @@ public final PutResponse<T> createOrUpdateInternal(UriInfo uriInfo, T updated) {
 
   @SuppressWarnings("unused")
   protected void postCreate(T entity) {
+    if (supportsSearchIndex) {
+      try {
+        searchClient.updateSearchEntityCreated(entity.getEntityReference());
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
     // Override to perform any operation required after creation.
     // For example ingestion pipeline creates a pipeline in AirFlow.
   }
 
   @SuppressWarnings("unused")
-  protected void postUpdate(T entity) {
-    // Override to perform any operation required after an entity update.
-    // For example ingestion pipeline creates a pipeline in AirFlow.
+  public void postUpdate(T entity) {
+    if (supportsSearchIndex) {
+      try {
+        if (entity.getEntityReference().getType().equals(Entity.TEST_SUITE)) {
+          searchClient.addTestCaseFromLogicalTestSuite(entity);
+        } else {
+          String scriptTxt = "for (k in params.keySet()) { ctx._source.put(k, params.get(k)) }";
+          searchClient.updateSearchEntityUpdated(entity.getEntityReference(), scriptTxt);
```

Comment:
@harshach , here I am finding .. handling TestCase and TestSuite is getting very tricky